mono/metadata/object.c

```diff
@@ -342,10 +343,9 @@ get_type_init_exception_for_vtable (MonoVTable *vtable)
 	 * in the hash.
 	 */
 	ex = NULL;
-	mono_domain_lock (domain);
-	if (domain->type_init_exception_hash)
-		ex = (MonoException *)mono_g_hash_table_lookup (domain->type_init_exception_hash, klass);
-	mono_domain_unlock (domain);
+	mono_mem_manager_lock (memory_manager);
```

Comment:
This looks problematic, a lot of code inside the lock might expect the domain lock to be held. It would be better to acquire both the domain and the memory manager lock.

Reply:
nm